Allow disabling cross-namespace references (#305)
fluxcd/image-automation-controller: docs/spec/v1beta1/imageupdateautomations.md

`GitRepository`, which is the default.

```go
// CrossNamespaceSourceReference contains enough information to let you locate the
// typed Kubernetes resource object at cluster level.
type CrossNamespaceSourceReference struct {
	// API version of the referent.
	// +optional
	APIVersion string `json:"apiVersion,omitempty"`

	// Kind of the referent.
	// +kubebuilder:validation:Enum=GitRepository
	// +kubebuilder:default=GitRepository
	// +required
	Kind string `json:"kind"`

	// Name of the referent.
	// +required
	Name string `json:"name"`

	// Namespace of the referent, defaults to the namespace of the Kubernetes resource object that contains the reference.
	// +optional
	Namespace string `json:"namespace,omitempty"`
}
```

### Cross-namespace references

> A ImageUpdateAutomationS  can refer to a GitRepository from a different namespace with 

Reviewer: ```suggestion
An `ImageUpdateAutomation`  can refer to a GitRepository from a different namespace with 
```

`spec.sourceRef.namespace` e.g.:

```yaml
apiVersion: image.toolkit.fluxcd.io/v1beta1
kind: ImageUpdateAutomation
metadata:

>   name: imageupdateautomation-sample

Reviewer: ```suggestion
  name: webapp
```

>   namespace: flux-system

Reviewer: ```suggestion
  namespace: apps
```

spec:
  interval: 5m
  sourceRef:
    kind: GitRepository # the only valid value, but good practice to be explicit here

>     name: sample-repo

Reviewer: ```suggestion
    name: apps
```

>     namespace: default

Reviewer: ```suggestion
    namespace: flux-system
```
